tsdb/store.go

```diff
@@ -535,14 +578,14 @@ func (s *Store) periodicMaintenance() {
 	}
 }
 
-// performMaintenance will loop through the shars and tell them
-// to perform any maintenance tasks. Those tasks should kick off
-// their own goroutines if it's anything that could take time.
+// performMaintenance loops through shards and executes any maintenance
+// tasks. Those tasks should run in their own goroutines if they will
+// take significant time.
 func (s *Store) performMaintenance() {
 	s.mu.Lock()
 	defer s.mu.Unlock()
-	for _, sh := range s.shards {
-		s.performMaintenanceOnShard(sh)
+	for _, sl := range s.shardLocations {
+		s.performMaintenanceOnShard(sl.Shard)
```

Comment:
I guess, yeah. To be honest a `shardLocation` shouldn't have a `nil` Shard, but better to be defensive :+1: